lib/msal-core/src/Configuration.ts

```diff
@@ -105,58 +132,13 @@ export type Configuration = {
  */
 
 // destructure with default settings
-export function buildConfiguration(
-  {
-    clientId = "",
-    authority = null,
-    validateAuthority = true,
-    redirectUri = () => Utils.getDefaultRedirectUri(),
-    postLogoutRedirectUri = () => Utils.getDefaultRedirectUri(),
-    state = "",
-    navigateToLoginRequestUrl = true
-  }: AuthOptions,
-  {
-    cacheLocation = "sessionStorage",
-    storeAuthStateInCookie = false
-  }: CacheOptions,
-  {
-    logger = new Logger(null),
-    loadFrameTimeout = FRAME_TIMEOUT,
-    tokenRenewalOffsetSeconds = OFFSET
-  }: SystemOptions,
-  {
-    isAngular = false,
-    unprotectedResources = new Array<string>(),
-    protectedResourceMap = new Map<string, Array<string>>()
-  }: FrameworkOptions
-): Configuration {
-  // restructure
-  let config: Configuration = {
-    auth: {
-      clientId,
-      authority,
-      validateAuthority,
-      redirectUri,
-      postLogoutRedirectUri,
-      state,
-      navigateToLoginRequestUrl
-    },
-    cache: {
-      cacheLocation,
-      storeAuthStateInCookie
-    },
-    system: {
-      logger,
-      loadFrameTimeout,
-      tokenRenewalOffsetSeconds
-    },
-    framework: {
-      isAngular,
-      unprotectedResources,
-      protectedResourceMap
-    }
+export function buildConfiguration({ auth, cache={}, system={}, framework={}}: Configuration): Configuration {
```

Comment:
The above code can cause linter error for us, cache = {} could be the right way. Minor but edit if you can.